src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Completion/MarkupTransitionCompletionItemProvider.cs

```diff
@@ -98,11 +98,17 @@ internal static bool AtMarkupTransitionCompletionPoint(RazorSyntaxNode owner)
                providing incorrect completion in the above two syntactically invalid
                scenarios.
             */
+            var encapsulatingMarkupElementNodeSeen = false;
```

Comment:
Not normally a fan of flag variables, but this seemed like a simple enough code path. Let me know if you'd like me to reconsider.

Reply:
I don't always mind but in this case it seems like this one in particular is hard to digest. The complexity it adds is mostly due to the rarity in seeing pieces of a System.Linq expression updating state and cross-checking that state.

To make this work and not violate some System.Linq crimes I'd suggest removing the `FirstOrDefault` and instead using a `foreach`.